### PR TITLE
frame system: add event util functions for tests.

### PR DESCRIPTION
**Pull request by shaunxw**
Add two functions for frame system events assertion, to make it easier to check expected events in unit tests and benchmarking.
- `assert_has_event` asserts the given `event` exists.
- `assert_last_event` asserts the last event is equal to the given one.

### CODE REVIEW

File: frame/balances/src/tests_reentrancy.rs [hunk 1]
```diff
@@ -189,23 +185,9 @@ fn transfer_dust_removal_tst1_should_work() {
...
-			assert!(
-				System::events().iter().any(
-					|er|
-					er.event == Event::pallet_balances(
-						crate::Event::DustLost(2, 50)
-					),
-				),
-			);
+			System::assert_has_event(Event::pallet_balances(crate::Event::Transfer(2, 3, 450)));
+
```
Comment (kianenigma) [inline frame/balances/src/tests_reentrancy.rs:189]:
```suggestion
```

File: frame/balances/src/tests_reentrancy.rs [hunk 2]
```diff
@@ -236,23 +218,9 @@ fn transfer_dust_removal_tst2_should_work() {
...
-			assert!(
-				System::events().iter().any(
-					|er|
-					er.event == Event::pallet_balances(
-						crate::Event::DustLost(2, 50),
-					),
-				),
-			);
+			System::assert_has_event(Event::pallet_balances(crate::Event::Transfer(2, 1, 450)));
+
```
Comment (kianenigma) [inline frame/balances/src/tests_reentrancy.rs:222]:
```suggestion
```

**Review by kianenigma — APPROVED** [review]
much appreciated.

**Review by thiolliere — APPROVED** [review]

**Comment by thiolliere** [comment]
bot merge

**Comment by shawntabrizi** [comment]
awesome, thanks :)

**Pull request merged**